dev/tools/lib/version.dart

```diff
@@ -0,0 +1,189 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+enum VersionType {
+  // Of the form x.y.z
+  stable,
+  // Of the form x.y.z-m.n.pre
+  development,
+  // Of the form x.y.z-m.n.pre.commits
+  latest,
+}
+
+class Version {
```

Comment:
This class could benefit from some docs or links to the existing version format docs

Reply:
done